oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profile/analyze/ProfileStackNode.java

```diff
@@ -0,0 +1,256 @@
+                ProfileStackNode node = it.next();
+                if (node != null && node.matches(childrenNode)) {
+                    childrenNode.combineDetectedStacks(node);
+                    continueChildrenMerging.accept(new Pair<>(childrenNode, node));
+
+                    it.set(null);
+                    break;
+                }
+            }
+        }
+
+        for (ProfileStackNode node : beingMergedNode.children) {
+            if (node != null) {
+                targetNode.children.add(node);
+            }
+        }
+    }
+
+    /**
+     * build GraphQL result, calculate duration and count data using parallels
+     * @return
+     */
+    public ProfileStackElement buildAnalyzeResult() {
+        // all nodes add to single-level list (such as flat), work for parallel calculating
+        LinkedList<Pair<ProfileStackElement, ProfileStackNode>> nodeMapping = new LinkedList<>();
+        ProfileStackElement root = buildElement();
+        nodeMapping.add(new Pair<>(root, this));
+
+        // same with combine logic
+        LinkedList<Pair<ProfileStackElement, ProfileStackNode>> stack = new LinkedList<>();
+        stack.add(new Pair<>(root, this));
+        while (!stack.isEmpty()) {
+            Pair<ProfileStackElement, ProfileStackNode> mergingPair = stack.pop();
+            ProfileStackElement respElement = mergingPair.key;
+
+            // generate children node and add to stack and all node mapping
+            respElement.setChildren(mergingPair.value.children.stream().map(c -> {
+                ProfileStackElement element = c.buildElement();
+                Pair<ProfileStackElement, ProfileStackNode> pair = new Pair<>(element, c);
+                stack.add(pair);
+                nodeMapping.add(pair);
+
+                return element;
+            }).collect(Collectors.toList()));
+        }
+
+        // calculate durations
+        nodeMapping.parallelStream().forEach(t -> t.value.calculateDuration(t.key));
+        nodeMapping.parallelStream().forEach(t -> t.value.calculateDurationExcludeChild(t.key));
+
+        return root;
+    }
+
+    private void detectedBy(ProfileStack stack) {
+        this.detectedStacks.add(stack);
+    }
+
+    private void combineDetectedStacks(ProfileStackNode node) {
+        this.detectedStacks.addAll(node.detectedStacks);
+    }
+
+    private ProfileStackElement buildElement() {
+        ProfileStackElement element = new ProfileStackElement();
+        element.setCodeSignature(this.codeSignature);
+        element.setChildren(new LinkedList<>());
+        element.setCount(this.detectedStacks.size());
+        return element;
+    }
+
+    /**
+     * calculate duration to {@link ProfileStackElement#getDuration()}
+     */
+    private void calculateDuration(ProfileStackElement element) {
+        if (this.detectedStacks.size() <= 1) {
+            element.setDuration(0);
+            return;
+        }
+
+        // sort owners
```

Comment:
`owners` -> `detectedStacks`, but I think the code itself is documentation already

Reply:
Ok, I understand. I will delete this comment